---
title: Short course Multi-trait <br> LAPEA
#subtitle: 
author: "Mateus Gonçalves"
bibliography: references.bib
date: "03/17/2025"
date-format: "DD/MM/YYYY"
format:
  revealjs:
    width: 1600
    height: 800
    title-slide-attributes:
      data-background-size: contain
      class: "no-footer-slide"  # Add a custom class to the title slide
    css: styles.css
    logo: Images\lapea_logo.jpeg
   # footer: "Code available on #[GitHub](https://github.com/goncalvesmateu/Multi-trait-LAPEA.git)."
    self-contained: true
    scrollable: false
    theme: default # built-in themes for reveal presentations
    background-transition: none
    progress: false # Show a progress bar at the bottom (defaults to true).
    history: true
    toc: false
    transition: slide
    transition-speed: default
    highlight-style: github # specify the code highlighting style
    code-line-numbers: true
    code-copy: hover
    controls: true # navigation arrows per slide
    navigation-mode: linear
    touch: true
    code-fold: true
    multiplex: false
    chalkboard: false
    slide-level: 2
    scroll-view: true
    slide-number: true
    show-slide-number: all
    callout-appearance: simple
    preview-links: auto
    incremental: false
editor: visual
---

## Outline

::: {style="font-size: 70%;"}
### [Background] {style="color:black"}

- [Correlation between traits]

- [Genetic correlation]

- [Response to selection]

- [Correlated response to selection]

### [Multiple trait selection] {style="color:black"}

-   [Tandem selection]
-   [Independent culling levels]
-   [Index selection]

### [Multi-trait model] {style="color:black"}

- [Uni-trait model]
- [Multi-trait model]
- [Hands-on using R]
:::

## Background 

::: {layout="[ 0.8, -0.08, 0.5 ]"}

-   Breeders select for multiple traits (morphological, physiological, biotic resistance,..,etc)

-   The list of traits can be endless, each with difference importance

-   Program goals, selection cycle, breeding (or generation) cycle


![**Fig 1**. *Model or ideal plant (IDEOTYPE) with attributes combined to maximize yield* <br> [@furbank2019field].](Images\ideotypeWheat.png){fig-align="left"}
:::

## Correlation between traits {style="font-size: 90%;"}

::: {style="font-size: 77%;"}
- [Phenotypic correlation]{style="color:red"}: linear association between the phenotypic values of two traits.

- [Genetic correlation]{style="color:red"}: linear association between the genetic (breeding) values of two traits. 

- [Environmental correlation]{style="color:red"}: linear association between the non-additive genetic, and nongentic effects of two traits. 


$$r_{P} = \frac {{COV}_{Pxy}}{{\sigma}_{Px}{\sigma}_{Py}}$$

::: {.fragment .fade-in}
$${{COV}_{Pxy}} = r_{P} ({\sigma}_{Px}{\sigma}_{Py})$$
:::

::: {.fragment .fade-in}
$${{COV}_{Pxy}} = {{COV}_{Gxy}} + {{COV}_{Exy}}$$
:::

::: {.fragment .fade-in}
\begin{align}
   r_{P} ({\sigma}_{Px}{\sigma}_{Py}) &= r_{G} ({\sigma}_{Gx}{\sigma}_{Gy}) + r_{E} ({\sigma}_{Ex}{\sigma}_{Ey}) \\
   & \vdots \\
  r_{P} &= {h}_{x}{h}_{y}r_{G} + {e}_{x}{e}_{y}r_{G} \\
\end{align}
:::

::: {.fragment .fade-in}
- [Phenotypic correlation results from the combination of genetic and enviromental correlations.]{style="color:red"} 

- [If both traits have low heritabilities then the phenotypic correlation is mainly due to enviromental correlations.]{style="color:red"}
:::

:::

## Genetic correlation

- Greatest interest for breeders

- Positive or negative and favorable or unfavorable, depending on goals. [Ex 1]{style="color:red"}: plant height and flowering date in barley are positively correlated. This correlation is favorable if harvest grains is the goal (short and early-maturing). But for forage (tall and late-maturing) is unfavorable [@bernardo2002breeding].

- Causes
  - [Linkage]{style="color:red"}: genes located on the same chromosome are genetically linked and do not segregate independently, like genes located on different chromosomes (can be dissipated by cycles of meiosis).

  - [Pleiotropy]{style="color:red"}: two traits are controlled by the same gene (can not be dissipated, more permanent).
  
## Response to selection

The mean genetic change in the trait of interest in a population.

### Direct response to selection

The selection criteria is based on the trait(s) of interest in the target environment. 

::: {layout="[ 0.4, -0.05, 0.6 ]"}

$$Rx = {k}_{p} {h}_{x}{\sigma}_{a}x$$
$$Ry = {k}_{p} {h}_{y}{\sigma}_{a}y$$

${k}_{p}$ is the selection differential when the proportion of selcted individuals is $p$

:::
  
## Indirect selection

- The criteria is based on trait(s) that may be associated with the trait(s) of interest (**secondary traits**). [Ex]{style="color:red"}: number of stalks and tons of cane per hectare. Also, the same trait measured in different environments[@rutkoski2019practical]:

::: {layout="[ 0.5, -0.05, 0.5 ]"}

![**Fig 2**. *Yield of maize in Tropical environment*](Images\cornTrop.jpg){fig-align="center" width=60%}


![**Fig 3**. *Yield of maize in temparate environment*](Images\cornTemp.png){fig-align="center" width=60%}
:::

- Indirect selection is often used because a secondary trait is easier and/or cheaper to evaluate, and measurable on both sex.

## Correlated response to selection

- [Correlated response to selection]{style="color:red"}: selection for one trait will cause a correlated response in the other (if genetic correlation exists).

### Indirect response to selection

$$Ry = {k}_{p} {h}_{y}{\sigma}_{a}y$$

- The superiority of indirect over direct selection depends the heritability of secondary trait and the genetic correlation, and when much larger population sizes are possible with indirect selection compared to direct selection.

## Multiple trait selection {.center}

Examples of multiple-trait selection strategies breeders employ... 

## Tandem selection
::: {style="font-size: 85%;"}
Selection for one trait until that trait is improved, then for a second, etc., until each has been improved to the desired level. Applicable in recurrent selection programs. [Ex]{style="color:red"}: tropical maize populations used as breeding material in temperate regions are first selected for photoperiod insensivity prior to selection for other traits.
:::

::: columns 
::: {.column width="50%"}
```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.8,
     xlab="Trait A", ylab="Trait B",col=ifelse(y >5, 'red', 'blue'))
title(main = "SELECT FOR TRAIT B FOR SEVERAL CYCLES",  cex.main = 2.5, col.main= "darkgreen")
abline(h=5, col="black", lwd = 2, lty = 2)

```
:::

::: {.column width="50%"}
```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.8,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10, 'red', 'blue'))
title(main = "SELECT FOR TRAIT A FOR SEVERAL CYCLES",  cex.main = 2.5, col.main= "darkgreen")
abline(v=10, col="black", lwd = 2, lty = 2)

```

:::

:::

## Independent culling levels {style="font-size: 90%;"}

A certain level of merit (minimun level of performance) is established for each trait, and all individuals below that level are discarded regardless of values for other traits.

```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.8,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10 & y >5, 'red', 'blue'))
abline(v=10, h=5, col="black", lwd = 2, lty = 2)

```

## Index selection

Select for $n$ traits simultaneously by using some index of net merit:

$$I = b_{1}X_{1} + b_{1}X_{1} + ... + b_{n}X_{n}$$
$b_{1}$ is the weight for trait $i$ and $X_{1}$ is the phenotypic value for trait $i$. 

Example adapted from [@bernardo2002breeding]: <br>
Price of 60 kg bag of maize: R$ 90,00 <br>
To dry a 60 kg bag of maize: R$ 0,25<br>
Target moisture concentration: 13%

A new candidate maize hybrid genotype has a yield of 100 bags of 60kg/ha and has a moisture of 15%. Then, the profit can be expressed in the form of the following selection index:

$$I = (90)(Yield) - (0,25)(Moisture - 13\%)(Yield)$$

## Uni-trait model

The BLUP methodology uses mixed models for the genetic analysis and provides accurate and least biased prediction of breeding values.

### Mixed model formulation 
$$
y = \mathbf{X}b + \mathbf{Z}u + e
$$
$y$: is an (nx1) vector of observations (phenotypes) <br> 
$b$: is an (px1) vector with fixed effects (e.g. environment)  <br>
$u$: is an (qx1) vector with random effects of breeding values; $u \sim N(\mathbf{0},\mathbf{K} \sigma_{g}^2)$ <br>
$e$: is an (nx1) vector with random effects of residuals; $e \sim N(\mathbf{0},\mathbf{R} \sigma_{e}^2)$ <br>
$\mathbf{X}$ and $\mathbf{Z}$: are design matrices relating obs to fixed and random effects, respectively.

## Uni-trait model

$$
y = \mathbf{X}b + \mathbf{Z}u + e
$$
\begin{align}

\begin{bmatrix}
   u \\
   e \\
   \end{bmatrix} \sim MVN
   
   \begin{pmatrix}
   \begin{bmatrix}
   0 \\
   0 \\
   \end{bmatrix} 
   ,
   \begin{bmatrix}
   \mathbf{K} & 0 \\
   0 & \mathbf{R} \\
    \end{bmatrix}
   \end{pmatrix}
\end{align}

\begin{align}
var(u)& = K&  \\
var(e)& = R& \\
var(y)& =ZGZ’ + R \\ 
E(y)& = Xb&
\end{align}

## Advantages

- Allows the analysis of unbalanced data

- Exploit information from relatives

Furthermore, it can be extended to incorporate/account for more effects, such as:

- Rows, columns, environments, permanent environmental... 

- Interaction terms (e.g GXE)

- Covariance and correlation structures (relationship matrices and Correlation matrices)

- Heterogeneous variances

- Correlated traits


## Multi-trait model


\begin{gather*}
y_{i} = X_{i}b_{i} + Z_{i}u_{i} + e_{i} , & i = 1, 2
\end{gather*}

\begin{align}

\begin{bmatrix}
   y_{1} \\
   y_{2} \\
   \end{bmatrix} = 
   
   \begin{bmatrix}
   X_{1} & 0 \\
   0 & X_{2} \\
   \end{bmatrix} \begin{bmatrix}
   b_{1} \\
   b_{2} \\
   \end{bmatrix} +
   
    \begin{bmatrix}
   Z_{1} & 0 \\
   0 & Z_{2} \\
   \end{bmatrix}\begin{bmatrix}
   u_{1} \\
   u_{2} \\
   \end{bmatrix} +
   
   \begin{bmatrix}
   e_{1} \\
   e_{2} \\
   \end{bmatrix}
   
\end{align}

## Advantages

- The prediction accuracy of low-heritability, difficult, and/or expensive to measure traits can be increased by using multi-trait models when the degree of correlation between traits is at least moderate,  to improve the target trait or all the correlated traits simultaneously (Calus and Veerkamp 2011). 

- Multi-trait models can be useful for increasing prediction accuracy when the traits of interest are not measured in the individuals of the testing set, but this and other traits were observed in individuals in the training set (Pszczola et al. 2013, Jia and Jannink, 2012).

## Hands-on using R

::: columns 
::: {.column width="40%"}
- [Chapter 5](https://link.springer.com/chapter/10.1007/978-3-030-89010-0_5#Abs1)

![[@montesinos2022multivariate]](Images\bookmulti.jpg){width=55% preview-link="true"}
:::

::: {.column width="60%"}
-   [Material on GitHub](https://github.com/goncalvesmateu/Multi-trait-LAPEA){preview-link="true"} 

``` tex
 Installing packages
 
 # Github version
 install.packages('devtools');
 library(devtools);
 install_github('covaruber/sommer')
 
 # or
 
 # CRAN version
 install.packages('sommer',dependencies = TRUE))
 library(sommer)
```
:::

:::

## Hands-on using R

``` tex
# rrBLUB (single-trait and no marker information used)

# GBLUP (single-trait with marker information used)

# rrBLUP (multi-trait and no marker information used)

# GBLUP (multi-trait with marker information used)



```


## References