---
title: Short course Multi-trait <br> LAPEA
#subtitle: 
author: "Mateus Gonçalves"
bibliography: references.bib
date: "03/17/2025"
date-format: "DD/MM/YYYY"
format:
  revealjs:
    width: 1600
    height: 800
    title-slide-attributes:
      data-background-size: contain
      class: "no-footer-slide"  # Add a custom class to the title slide
    css: styles.css
    logo: Images\lapea_logo.jpeg
   # footer: "Code available on #[GitHub](https://github.com/goncalvesmateu/Multi-trait-LAPEA.git)."
    self-contained: true
    scrollable: false
    theme: default # built-in themes for reveal presentations
    background-transition: none
    progress: false # Show a progress bar at the bottom (defaults to true).
    history: true
    toc: false
    transition: slide
    transition-speed: default
    highlight-style: github # specify the code highlighting style
    code-line-numbers: true
    code-copy: hover
    controls: true # navigation arrows per slide
    navigation-mode: linear
    touch: true
    code-fold: true
    multiplex: false
    chalkboard: false
    slide-level: 2
    scroll-view: true
    slide-number: true
    show-slide-number: all
    callout-appearance: simple
    preview-links: auto
    incremental: false
editor: visual
---

## Outline

::: {style="font-size: 75%;"}
### [Background] {style="color:black"}

- [Phenotypic, genetic and environmental correlation]

- [Direct and indirect selection]

### [Multiple trait selection] {style="color:black"}

-   [Multistage selection]
-   [Tandem selection]
-   [Independent culling levels]
-   [Index selection]

### [Multi-trait model] {style="color:black"}

- [Uni-trait model]
- [Multi-trait model]
- [Hands-on using R]
:::

## Background 

::: {layout="[ 0.8, -0.3, 0.5 ]"}

-   Breeders select for multiple traits (morphological, physiological, disease resistance...)

-   The list of traits can be endless
Depending on the program or project goals, one of three types of multiple-trait selection can be employed. A brief description of these types follows.


![**Fig 1**. *Model or ideal plant with attributes <br> combined to maximize yield* <br> [@furbank2019field].](Images\ideotypeWheat.png){fig-align="left"}
:::

## Phenotypic, genetic and environmental correlation {style="font-size: 90%;"}


```{mermaid}{fig-width: 10.5}
flowchart LR
  A[Hard edge] --> B(Round edge)
  B --> C{Decision}
  C --> D[Result one]
  C --> E[Result two]
```


## Genetic correlation

Genes located on the same chromosome are genetically linked. Genetic linkage analysis can be used to determine the order of genes on chromosomes. Closely linked genes are not segregating independently, like genes located on different chromosomes. This has different implications, e.g., in relation to trait correlations

$$
\text{Classes} =  \begin{cases}
               \bar{c} - (\tau)  sd(c) < \hat{c}_v < \bar{c} + (\tau)  sd(c) \rightarrow \text{Homoeostatic} \\ 
                \hat{c}_v > \bar{c} + (\tau) sd(c) \rightarrow \text{Sensitive} \\ 
                \hat{c}_v < \bar{c} - (\tau)  sd(c) \rightarrow \text{Aggresive}               
            \end{cases}
$$

## Direct and indirect selection

In direct selection, the selection criteria is based on the trait(s) of
interest. For example, if the trait of interest is grain yield in the target environment, selections are made using grain yield data from trials conducted in
the target environment. 

In indirect selection, the criteria is based on traits that may be associated with
grain yield such as biomass yield, grain size, number of inflorescences, etc.,
would be considered indirect selection, and such traits are called **secondary
traits**. Likewise, selection based on the trait of interest measured in an environment that does not represent the target environment can be considered
indirect selection: For instance, if the target environment is tropical, grain yield measured in a temperate environment would be considered a secondary trait and selection based on this trait would be considered indirect selection. 

Genomic selection is a form of indirect selection because selection is based on a genotype and not directly on phenotypic values.
Indirect selection is often used to eliminate a fraction of the selection
units prior to selection on the trait of interest especially when the trait of
interest is expensive to measure. For example, in rice breeding, the cost
of conducting sensory evaluations to assess grain quality is so high, quality
can only be measured on relatively few evaluation units. 

To determine if indirect selection would be more effective than indirect
selection, a metric called The relative efficiency of indirect vs. direct selection is simply the ratio of the expected genetic gain due to indirect selection to the expected gain due to direct selection (Lerner and Cruden, 1947). 

Assuming the population
sizes and number of individuals selected is constant, the relative efficiency
of indirect and direct selection depends the selection accuracy of indirect
selection as well as the genetic correlation between the indirect and direct
selection criteria (Fig. 3). For the case of indirect selection based on markers,
indirect selection accuracy equals one, because a marker genotype is
completely heritable. When much larger population sizes are possible with indirect selection compared to direct selection, there is a greater chance that
indirect selection would be a more favorable strategy.

## Multiple trait selection {.center}

Examples of multiple-trait selection strategies can be employed... 

## Multistage selection

Multistage selection: Selection for different traits at different stages during cultivar development.

```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.5,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10 & y >5, 'red', 'blue'))
abline(v=10, h=5, col="black", lwd = 2, lty = 2)

```

## Tandem selection

Selection for one trait until that trait is improved, then for a second, etc., until finally each has been improved to the desired level.

```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.5,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10 & y >5, 'red', 'blue'))
abline(v=10, h=5, col="black", lwd = 2, lty = 2)

```

## Independent culling levels

A certain level of merit (minimun level of performance) is established for each trait, and all individuals below that level are discarded regardless of values for other traits.

```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.5,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10 & y >5, 'red', 'blue'))
abline(v=10, h=5, col="black", lwd = 2, lty = 2)

```

## Index selection

Select for all traits simultaneously by using some index of net merit. 

```{r fig.align='center', fig.cap="", fig.height = 6}
#| echo: true

set.seed(1234)
x <- rnorm(200, mean = 10, sd = 4)
y <- rnorm(200, mean = 6, sd = 2)

dat <- data.frame(x,y)

# Visualize the scatter plot of traits A and B
plot(x,y, pch=19, ylim = c(min(dat$y), max(dat$y)),xlim = c(min(dat$x), max(dat$x)),
     rect(par("usr")[1], par("usr")[3],
          par("usr")[2], par("usr")[4], col = "lightgrey"),
     cex.lab=1.5, cex.axis=1.5,
     xlab="Trait A", ylab="Trait B",col=ifelse(x > 10 & y >5, 'red', 'blue'))
abline(v=10, h=5, col="black", lwd = 2, lty = 2)

```

## Another slide 3

::: columns
::: {.column width="40%"}
``` tex
\begin{gather*}
a_1=b_1+c_1\\
a_2=b_2+c_2-d_2+e_2
\end{gather*}

\begin{align}
a_{11}& =b_{11}&
  a_{12}& =b_{12}\\
a_{21}& =b_{21}&
  a_{22}& =b_{22}+c_{22}
\end{align}
```
:::

::: {.column width="60%"}
```{=tex}
\begin{gather*}
a_1=b_1+c_1\\
a_2=b_2+c_2-d_2+e_2
\end{gather*}
```
```{=tex}
\begin{align}
a_{11}& =b_{11}&
  a_{12}& =b_{12}\\
a_{21}& =b_{21}&
  a_{22}& =b_{22}+c_{22}
\end{align}
```
:::
:::

## Uni-trait model

The BLUP methodology uses mixed models for the genetic analysis and provides accurate and least biased prediction of breeding values.

### Mixed model formulation 
$$
y = Xb + Zu + e
$$

::: {layout="[ 0.55, -0.05, 0.45 ]"}
$y$: is the vector of observations (phenotypes) <br> 
$b$: is the vector with fixed effects with design matrix  <br>
$u$: is the vector with random effects with design matrix <br>
$X$ and $Z$: are design matrices relating obs to fixed and random effects, respectively.

\begin{align}
u \sim N(0,\sigma_{1}) & e \sim N(0,\sigma_{1}) & a_{12}& =b_{12}\\
E(y) = Xb & var(u)= G&\\
  var(e)= R & var(y)=ZGZ’ + R
\end{align}
:::

## Advantages

- It handles missing data

## Advantages

Furthermore, it can be extended to incorporate/account for more complicated effects, such as:

- Interaction terms (e.g GXE)

- Covariance and correlation structures (relationship matrices and Correlation matrices)

- Heterogeneous variances

- Correlated traits


## Multi-trait model


\begin{gather*}
y_{i} = X_{i}b_{i} + Z_{i}u_{i} + e_{i} , & i = 1, 2
\end{gather*}

\begin{align}

\begin{bmatrix}
   y_{1} \\
   y_{2} \\
   \end{bmatrix} = 
   
   \begin{bmatrix}
   X_{1} & 0 \\
   0 & X_{2} \\
   \end{bmatrix} \begin{bmatrix}
   b_{1} \\
   b_{2} \\
   \end{bmatrix} +
   
    \begin{bmatrix}
   Z_{1} & 0 \\
   0 & Z_{2} \\
   \end{bmatrix}\begin{bmatrix}
   u_{1} \\
   u_{2} \\
   \end{bmatrix} +
   
   \begin{bmatrix}
   e_{1} \\
   e_{2} \\
   \end{bmatrix}
   
\end{align}

## Advantages

In this and many other cases, sometimes the interest is to predict traits that are difficult or expensive to measure with those that are easy to measure or the aim can be to improve all these correlated traits simultaneously (Henderson and Quaas 1976; Calus and Veerkamp 2011; Jiang et al. 2015). In these lines, there is evidence of the usefulness of multi-trait modelling. For example, Jia and Jannink (2012) showed that, compared to single-trait modelling, the prediction accuracy of low-heritability traits could be increased by using a multi-trait model when the degree of correlation between traits is at least moderate. They also found that multi-trait models had better prediction accuracy when phenotypes were not available on all individuals and traits. Joint modelling also has been found useful for increasing prediction accuracy when the traits of interest are not measured in the individuals of the testing set, but this and other traits were observed in individuals in the training set (Pszczola et al. 2013).

## Hands-on using R

::: columns 
::: {.column width="40%"}
- [Chapter 5](https://link.springer.com/chapter/10.1007/978-3-030-89010-0_5#Abs1)

![[@montesinos2022multivariate]](Images\bookmulti.jpg){width=55%}
:::

::: {.column width="60%"}
-   [Material on GitHub](https://github.com/goncalvesmateu/Multi-trait-LAPEA) 

``` tex
 Installing packages
 
 # Github version
 install.packages('devtools');
 library(devtools);
 install_github('covaruber/sommer')
 
 # or
 
 # CRAN version
 install.packages('sommer',dependencies = TRUE))
 library(sommer)
```
:::

:::

## Hands-on using R

``` tex
# rrBLUB (single-trait and no marker information used)

# GBLUP (single-trait with marker information used)

# GBLUP (multi-trait and no marker information used)

# GBLUP (multi-trait with marker information used)



```


## References